Python/pylifecycle.c

```diff
@@ -1348,6 +1351,38 @@ Py_FinalizeEx(void)
     PyThreadState *tstate = _PyRuntimeState_GetThreadState(runtime);
     PyInterpreterState *interp = tstate->interp;
 
+    /* Check we're running in the main interpreter (not yet supported to call
+     * from any interpreter).
+     */
+    if (interp != PyInterpreterState_Main()) {
+        fprintf(stderr,
+                "Py_FinalizeEx: error: must be called from the main interpreter\n");
+        return -1;
```

Comment:
I consider this as a programming bug and maybe Py_FatalError() would be more appropriate here.